packages/wallet-sdk/src/sign/scw/utils/createSubAccountSigner.ts

```diff
@@ -52,74 +70,198 @@ export async function createSubAccountSigner({ chainId }: { chainId: number }) {
-        case 'net_version':
-          return chainId.toString();
-        case 'eth_chainId':
-          return numberToHex(chainId);
-        case 'eth_sendTransaction': {
-          assertArrayPresence(args.params);
-          return account.sign(args.params[0] as { hash: Hex });
+  const request: EIP1193RequestFn<
+    [...WalletRpcSchema, SendPreparedCallsSchema, PrepareCallsSchema]
+  > = (async (args) => {
+    switch (args.method) {
+      case 'wallet_addSubAccount':
+        return subAccount!;
+      case 'eth_accounts':
+        return [subAccount!.address] as Address[];
+      case 'eth_coinbase':
+        return subAccount!.address;
+      case 'net_version':
+        return chainId.toString();
+      case 'eth_chainId':
+        return numberToHex(chainId);
+      case 'eth_sendTransaction': {
+        assertArrayPresence(args.params);
+        return account.sign(args.params[0] as { hash: Hex });
+      }
+      case 'wallet_sendCalls': {
+        assertArrayPresence(args.params);
+        // Get the client for the chain
+        const chainId = get(args.params[0], 'chainId') as number;
+        assertPresence(chainId, standardErrors.rpc.invalidParams('chainId is required'));
+
+        const walletClient = getClient(chainId);
+        assertPresence(walletClient, standardErrors.rpc.internal('wallet client not found'));
+
+        if (!args.params[0]) {
+          throw standardErrors.rpc.invalidParams('params are required');
+        }
+
+        if (!('calls' in args.params[0])) {
+          throw standardErrors.rpc.invalidParams('calls are required');
+        }
+
+        const prepareCallsResponse = await request({
+          method: 'wallet_prepareCalls',
+          params: [
+            {
+              calls: args.params[0].calls as {
+                to: Address;
+                data: Hex;
+                value: Hex;
+              }[],
+              chainId: numberToHex(chainId),
+              from: subAccount.address,
+              capabilities:
+                'capabilities' in args.params[0]
+                  ? (args.params[0].capabilities as Record<string, any>)
+                  : {},
+            },
+          ],
+        });
+
+        const signature = await owner!.sign?.({
+          // Hash returned from wallet_prepareCalls is double hex encoded
+          hash: hexToString(prepareCallsResponse.signatureRequest.hash) as `0x${string}`,
+        });
+
+        let signatureData: SendPreparedCallsSchema['Parameters'][0]['signature'];
+
+        if (!signature) {
+          throw standardErrors.rpc.internal('signature not found');
+        }
+
+        if (owner!.address && isAddress(owner!.address)) {
+          signatureData = {
+            type: 'secp256k1',
+            data: {
+              address: owner!.address,
+              signature: signature as Hex,
+            },
+          };
+        } else if (typeof signature === 'object' && 'webauthn' in signature) {
```

Comment:
```suggestion
        } else if (!get(signature, 'webauthn')) {
```

Reply:
hm seems like `get` doesn't actually narrow the type

Reply:
managed to avoid it entirely